### PR TITLE
Added test for RCS1196

### LINKED ISSUE #1128: Redundant parentheses in "Call extension method as instance method"
**Product and Version Used**: 4.3.0

**Steps to Reproduce**:

Before:
```cs
WriteValue(writer, value);
```
After:
```cs
(writer).WriteValue(value);
```

### PR DESCRIPTION
**Pull request by jakubreznak**
added test for RCS1196
resolve #1128

I tested it and there were no redundant parentheses. I also wrote test for it.

### CODE REVIEW

**Comment by josefpihrt** [comment]
I took a look at it and I turned out that the problem is with the refactoring which has the same functionality as the analyzer:

https://github.com/dotnet/roslynator/blob/d52ec5773ef7404ac406b6d3bc2860c2752a10b8/src/Refactorings/CSharp/Refactorings/InvocationExpressionRefactoring.cs#L55-L64

**Comment by josefpihrt** [comment]
@jakubreznak Could you fix it?

**Comment by jakubreznak** [comment]
Yeah, sorry, i will look at it, but I am not able to do it this week.

**Comment by jakubreznak** [comment]
I looked at it and fixed it, also added a test. Thank you for your suggestion.

File: src/Common/CSharp/Analysis/CallExtensionMethodAsInstanceMethodAnalysis.cs [hunk 1]
```diff
@@ -110,14 +110,14 @@ MemberAccessExpressionSyntax CreateNewMemberAccessExpression()
                 case SyntaxKind.GenericName:
                     {
                         return SimpleMemberAccessExpression(
-                            ParenthesizedExpression(argument.Expression),
```
Comment (josefpihrt) [inline src/Common/CSharp/Analysis/CallExtensionMethodAsInstanceMethodAnalysis.cs:113]: I think that parentheses could be necessary sometimes. Better would be to keep parentheses and add `SimplifiedAnnotation` which will take care of removing parentheses if redundant. There is extension method `Parenthesize` which will parenthesize the expression and add `SimplifierAnnotation` by default.

File: src/Common/CSharp/Analysis/CallExtensionMethodAsInstanceMethodAnalysis.cs [hunk 2]
```diff
@@ -110,14 +110,14 @@ MemberAccessExpressionSyntax CreateNewMemberAccessExpression()
...
+                            argument.Expression,
                             (SimpleNameSyntax)expression);
                     }
                 case SyntaxKind.SimpleMemberAccessExpression:
                     {
                         var memberAccess = (MemberAccessExpressionSyntax)expression;
 
-                        return memberAccess.WithExpression(ParenthesizedExpression(argument.Expression));
```
Comment (josefpihrt) [inline src/Common/CSharp/Analysis/CallExtensionMethodAsInstanceMethodAnalysis.cs:120]: Some as previous comment.

**Review by josefpihrt — CHANGES_REQUESTED** [review]

File: ChangeLog.md [hunk 3]
```diff
@@ -30,6 +30,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Detect false positive from Unity code ([RCS1169](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1169)) ([PR](https://github.com/dotnet/roslynator/pull/1245))
   - Introduce config option `roslynator_unity_code_analysis.enabled = true|false`
   - Make option `roslynator_suppress_unity_script_methods` obsolete
+- Fix [RCS1196](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1196/) ([PR](https://github.com/dotnet/roslynator/pull/1235))
```
Comment (josefpihrt) [inline ChangeLog.md:33]: Please move to "Unreleased" section

File: src/Common/CSharp/Analysis/CallExtensionMethodAsInstanceMethodAnalysis.cs (hunk 1 above)
Comment (jakubreznak) [reply src/Common/CSharp/Analysis/CallExtensionMethodAsInstanceMethodAnalysis.cs:113]: Thank you, I put there the extension method Parenthesize.

**Review by josefpihrt — APPROVED** [review]

**Comment by josefpihrt** [comment]
@jakubreznak Thanks for the contribution!

**Pull request merged**